integrations/google_assistant_dialogflow/actions.md

```diff
@@ -0,0 +1,67 @@
+
+- Once the Action is created, go to [Dialogflow](https://dialogflow.cloud.google.com/) and create new Agent. At Google Project select the Action you created in Google Actions Console.
+
+- Now you have the Dialogflow Agent integrated with Assistant Action.
+
+- At Intents delete the Default Welcome Intent. Now create a new intent, called "welcome". Add the "Google Assistant Welcome" Event, and enable webhook call for this intent at the bottom section Fulfillment. Save intent.
+
+- Go to Integration section, enter in "Integration settings". Select "welcome" at Explicit invocation and click "Test". It updates your Action project for next step.
+
+- Now you have to create a Function, the backend that will be deploy at Google Cloud Functions. Run `firebase init` at you folder project. Select "Functions: Configure and deploy Cloud Functions", then "Use an existent project". Select the project id from the list. Select JavaScript or TypeScript language, Eslint and if you want to install dependencies. It creates a "functions" folder with the Funciton.
+
+- At "functions/index.js" insert this code below:
+
+```
+const functions = require('firebase-functions'),
+	{ dialogflow } = require('actions-on-google'),
+	app = dialogflow();
+
+app.intent('welcome', conv => conv.ask(`¡Bienvenido a tu Action de Assistant!`));  
+
+exports.dialogflowFirebaseFulfillment = functions.https.onRequest(app);
+```
+
+- Move to functions folder and install "actions-on-google" package with `npm install actions-on-google`.
+
+- Execute `firebase deploy --only functions:dialogflowFirebaseFulfillment` to deploy the code at Google Cloud Functions.
+
+- For some reason the deploy does not impact at first, you must go to Fulfillment at Dialogflow section, enable "Inline Editor", copy the same code at index.js and package.json from your function created before, and deploy in Dialogflow. When the deploy ends, you are able to change your first code at the folder you created first, and use the command `firebase deploy --only functions:dialogflowFirebaseFulfillment` to deploy it.
+
+- Go to Overview tab at [Google Actions Console](https://console.actions.google.com/) and complete the Quick setup, you have to complete the Assistant Action name by which it will be invoked.
+
+- Go to Test tab at [Google Actions Console](https://console.actions.google.com/) and start testing.
+
+## Pain Points
+
+- When you have to deploy you develop to Production at Assistant, you need to pass de Google Review. They test all the flows and intents of the action. If you neeed to enter a code or some id in the conversation and you get a dinamic answer, they demand to use [Account linking](https://developers.google.com/assistant/identity), cause they think you are using some external API or service. To use Account linking, you have to implement some login with Google Accounts or an external API with some web view.
```

Comment:
deploy you develop -> deploy the development
de Google Review -> the Google Review
neeed -> need
dinamic -> dynamic
Account linking -> Account Linking